doc/developer/design/20240508_subsource_schemas_after_planning.md

```diff
@@ -0,0 +1,113 @@
+# Restructure + genericize subsource planning
+
+-   Associated:
+    -   https://github.com/MaterializeInc/materialize/issues/20208
+    -   https://github.com/MaterializeInc/materialize/pull/26881
+    -   https://github.com/MaterializeInc/materialize/issues/26999
+
+## The Problems
+
+-   To complete #20208, we have to know the schema of the Kafka topic. This
```

Comment:
Why? I didn't see in the above doc why this is necessarily different from kafka sources today, where we also need to know their schema.

Reply:
`CreateSubsourceStatement` requires knowing the schema of the object.

Reply:
Sure but isn't that also true today? I'm perhaps misunderstanding your phrasing here -- how does that constraint block the implementation of #20208?

Reply:
maybe this phrasing is clearer?

> we have to know the schema of the Kafka topic when creating the subsource/table that it will write to.